src/main/java/com/faforever/client/patch/SimpleHttpFeaturedModUpdaterTask.java

```diff
@@ -54,32 +68,50 @@ protected PatchResult call() throws Exception {
 
     List<FeaturedModFile> featuredModFiles = fafService.getFeaturedModFiles(featuredMod, version).get();
+    Path fafDataDirectory = preferencesService.getFafDataDirectory();
 
     Path initFile = null;
+    Path cacheFilePath;
+    Path targetPath;
+
+    Map<String, String> knownTargetHashes = new HashMap<>();
+
+    // Download to cache if file exists in target place, otherwise, download to direct place.
     for (FeaturedModFile featuredModFile : featuredModFiles) {
-      Path fafDataDirectory = preferencesService.getFafDataDirectory();
-      Path targetPath = fafDataDirectory
-          .resolve(featuredModFile.getGroup())
-          .resolve(featuredModFile.getName());
-
-      if (Files.exists(targetPath)
-          && featuredModFile.getMd5().equals(com.google.common.io.Files.hash(targetPath.toFile(), Hashing.md5()).toString())) {
-        logger.debug("Already up to date: {}", targetPath);
+      targetPath = fafDataDirectory.resolve(featuredModFile.getGroup()).resolve(featuredModFile.getName());
+      cacheFilePath = featuredModFileCacheService.getCachedFilePath(featuredModFile);
+
+      String existingTargetFileHash = null;
+      if (Files.exists(targetPath)) {
+        existingTargetFileHash = com.google.common.io.Files.hash(targetPath.toFile(), Hashing.md5()).toString();
+        knownTargetHashes.put(targetPath.toString(), existingTargetFileHash);
+
+        if (!featuredModFile.getMd5().equals(existingTargetFileHash)) {
+          logger.info(String.format("downloading: %s", cacheFilePath.toString()));
+          downloadFeaturedModFile(featuredModFile, cacheFilePath);
+        }
       } else {
-        Files.createDirectories(targetPath.getParent());
-        updateMessage(i18n.get("updater.downloadingFile", targetPath.getFileName()));
-
-        String url = featuredModFile.getUrl();
-        downloadService.downloadFile(new URL(url), targetPath, this::updateProgress);
-        UpdaterUtil.extractMoviesIfPresent(targetPath, fafDataDirectory);
+        logger.info(String.format("downloading: %s", targetPath.toString()));
+        downloadFeaturedModFile(featuredModFile, targetPath);
+        knownTargetHashes.put(targetPath.toString(), featuredModFile.getMd5());
       }
 
-
       if ("bin".equals(featuredModFile.getGroup()) && initFileName.equalsIgnoreCase(featuredModFile.getName())) {
         initFile = targetPath;
       }
     }
 
+    for (FeaturedModFile featuredModFile : featuredModFiles) {
+      targetPath = fafDataDirectory.resolve(featuredModFile.getGroup()).resolve(featuredModFile.getName());
+      cacheFilePath = featuredModFileCacheService.getCachedFilePath(featuredModFile);
+      String existingTargetFileHash = knownTargetHashes.get(targetPath.toString());
+
+      if (!featuredModFile.getMd5().equals(existingTargetFileHash)) {
+        logger.info(String.format("copying featured mod file: %s to %s", cacheFilePath.toString(), targetPath.toString()));
```

Comment:
`logger.info("Copying featured mod file '{}' to '{}'", cacheFilePath, targetPath);`

Reply:
done